packages/cli-kit/src/public/node/api/graphql.ts

```
@@ -113,12 +146,35 @@ async function performGraphQLRequest<TResult>(options: PerformGraphQLRequestOpti
     }
   }
 
+  const unauthorizedHandlerFunction =
+    unauthorizedHandler ??
+    (async () => {
+      if (refreshTokenOnAuthorizedResponse) {
+        const refreshTokenResult = await refreshTokenOnAuthorizedResponse()
+        if (refreshTokenResult.action === 'retry') {
+          const {token: refreshedToken} = refreshTokenResult
+          const {client: newClient, headers: newHeaders} = await createGraphQLClient({
+            url,
+            addedHeaders,
+            token: refreshedToken,
+          })
+          client = newClient
+          headers = newHeaders
+          return {action: 'continue'}
+        } else {
+          return {action: 'throw'}
```

Should this `return refreshTokenResult` instead instead of hardcoding `throw`? The interface already expresses the allowed return types, and by hardcoding `throw` here we are making the assumption there will only ever be 2 possibilities, instead of just delegating that to the type (and allowing `retry` to be the special case in this component

---

Alternatively, is there value in reproducing the `{action: ...}` part on the `refreshTokenOnAuthorizedResponse` Function? What if that function just returns a new token or it doesn't, and then this code here could do the mapping: "If I get a new token, I continue, else I throw". 

It feels more complex to bleed the `UnauthorizedHandlerResult` type through multiple layers.